wasm/string.h

```diff
@@ -0,0 +1,33 @@
+/******************************************************************************
+ *
+ *  Copyright 2024 Google LLC
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at:
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ ******************************************************************************/
+
+#ifndef __LC3_STRING_H
+#define __LC3_STRING_H
+
+#include <stddef.h>
+
+static inline void *memcpy(void * restrict d, const void * restrict s, size_t n)
```

Comment:
nit: I would avoid this, as there is cases where `__builtin_memcpy` (Same for other builtins) is replaced with a call to memcpy instead.

And in this case since you defined memcpy, I'm not sure that we will get an undefined symbol.